### PR TITLE
feat(cli): consensus engine events

### LINKED ISSUE #2399: emit forkchoice events from consensus engine
### Describe the feature

for better observability, we need to emit events from the consensus engine, like FCU

so it can be consumed by the CLI event handler:

https://github.com/paradigmxyz/reth/blob/607186de716b5c2985142b60a3f3e52d142f003e/bin/reth/src/node/events.rs#L76-L83

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rkrasiuk**
Closes #2399 

Refactor current event emitter to use a common `EventListeners` implementation (not sure it belongs in `primitives`, but didn't find a better place).

Start emitting events in beacon consensus engine.

### CODE REVIEW

File: crates/net/network/src/manager.rs [hunk 1]
```diff
@@ -899,27 +899,3 @@ pub enum NetworkEvent {
...
-impl NetworkEventListeners {
-    /// Sends  the event to all listeners.
-    ///
-    /// Remove channels that got closed.
-    fn send(&mut self, event: NetworkEvent) {
-        self.listeners.retain(|listener| {
-            let open = listener.send(event.clone()).is_ok();
-            if !open {
-                trace!(target : "net", "event listener channel closed",);
-            }
```
Comment (Rjected) [inline crates/net/network/src/manager.rs:921]: do we think this trace is useful / worth keeping in the new impl?

**Review by Rjected — APPROVED** [review]
lgtm, primitives seems fine? I don't really see a better place to put it

File: crates/primitives/src/utils/event_listeners.rs
Comment (mattsse) [inline crates/primitives/src/utils/event_listeners.rs:1]: hmm not sure utils is the right location.
I just want be mindful that utils doesn't become a dumping ground for random things.

fine with moving this to a `listener` modul

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/primitives/src/utils/event_listeners.rs
Comment (rkrasiuk) [reply crates/primitives/src/utils/event_listeners.rs:1]: alright, sg. i bet it def would become a dumping ground

File: crates/net/network/src/manager.rs (hunk 1 above)
Comment (mattsse) [reply crates/net/network/src/manager.rs:921]: yeh, doesn't hurt imo

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by mattsse** [comment]
bench seems unrelated

**Pull request merged**